Source/Core/DolphinQt/Config/Mapping/GCPadWiiUConfigDialog.cpp

```diff
@@ -24,28 +24,31 @@ GCPadWiiUConfigDialog::GCPadWiiUConfigDialog(int port, QWidget* parent)
   ConnectWidgets();
 }
 
+GCPadWiiUConfigDialog::~GCPadWiiUConfigDialog()
+{
+  GCAdapter::SetAdapterCallback(nullptr);
+}
+
 void GCPadWiiUConfigDialog::CreateLayout()
 {
   setWindowTitle(tr("GameCube Adapter for Wii U at Port %1").arg(m_port + 1));
 
-  const bool detected = GCAdapter::IsDetected();
   m_layout = new QVBoxLayout();
-  m_status_label = new QLabel(detected ? tr("Adapter Detected") : tr("No Adapter Detected"));
+  m_status_label = new QLabel();
   m_rumble = new QCheckBox(tr("Enable Rumble"));
   m_simulate_bongos = new QCheckBox(tr("Simulate DK Bongos"));
   m_button_box = new QDialogButtonBox(QDialogButtonBox::Ok);
 
+  UpdateAdapterStatus();
+
+  GCAdapter::SetAdapterCallback(
+      [this] { QMetaObject::invokeMethod(this, &GCPadWiiUConfigDialog::UpdateAdapterStatus); });
```

Comment:
There's no need to use `QMetaObject::invokeMethod` here. You can simply put `UpdateAdapterStatus();` instead, as unqualified function names are implied to reference `this`.

Reply:
The callback will be called from a background thread. Is it safe to directly update the GUI from it?
(For the record, [the Wx code](https://github.com/dolphin-emu/dolphin/blob/227290055d6685281c72fd1d2076b67fdfb1b858/Source/Core/DolphinWX/Config/GCAdapterConfigDiag.cpp#L67) used `wxQueueEvent` to trigger the update on the main thread)

Reply:
I'd suggest using ``QueueOnObject`` from our ``QtUtils/QueueOnObject.h`` header file instead.